photon-api/src/main/scala/com/linkedin/photon/ml/util/DefaultIndexMapLoader.scala

```diff
@@ -48,18 +44,14 @@ object DefaultIndexMapLoader {
   /**
    * A factory method to create default index map loaders.
    *
-   * NOTE: the resulting indexes take their values in [0..numFeatures] ("distinct.sorted.zipWithIndex").
+   * @note The resulting indexes take their values in [0..numFeatures] ("distinct.sorted.zipWithIndex").
    *
    * @param sc The Spark context
    * @param featureNames A set of feature names
    * @return A default index map loader
    */
-  def apply(sc: SparkContext, featureNames: Seq[String]): DefaultIndexMapLoader = {
-    // TODO: "distinct.sorted.zipWithIndex" is duplicated in DefaultIndexMap. Remove one.
-    val indexMapLoader = new DefaultIndexMapLoader(featureNames.distinct.sorted.zipWithIndex.toMap)
-    indexMapLoader.prepare(sc) // need to call this here to complete initialization
-    indexMapLoader
-  }
+  def apply(sc: SparkContext, featureNames: Seq[String]): DefaultIndexMapLoader =
+    new DefaultIndexMapLoader(sc, featureNames.distinct.sorted.zipWithIndex.toMap)
```

Comment:
Yeah, it's probably okay for these cases, since we're only dealing with DefaultIndexMap here.